cupy/_core/core.pyx

```diff
@@ -60,6 +60,20 @@ from cupy.exceptions import ComplexWarning
 NUMPY_1x = numpy.__version__ < '2'
 
 
+cdef extern from *:
+    """
+    #define _str_(s) #s
+    #define _xstr_(s) _str_(s)
+    const char* cupy_cache_key = _xstr_(CUPY_CACHE_KEY);
+    #undef _xstr_
+    #undef _str_
```

Comment:
Am curious why all the indirection is needed here?

Reply:
`CUPY_CACHE_KEY` is a compiled-time variable that needs to stringification.